src/web/app/pages-instructor/instructor-search-page/instructor-search-page.component.ts

```diff
@@ -69,22 +73,25 @@ export class InstructorSearchPageComponent implements OnInit {
    * Searches for students and questions/responses/comments matching the search query.
    */
   search(searchQuery: SearchQuery): void {
+    this.loadingBarService.showLoadingBar();
     const paramMap: { [key: string]: string } = {
       searchkey: searchQuery.searchKey,
       searchstudents: searchQuery.searchStudents.toString(),
       searchfeedbacksessiondata: searchQuery.searchFeedbackSessionData.toString(),
     };
-    this.httpRequestService.get('/studentsAndSessionData/search', paramMap).subscribe((resp: SearchResult) => {
-      this.studentTables = resp.searchStudentsTables;
-      this.fbSessionDataTables = resp.searchFeedbackSessionDataTables;
-      const hasStudents: boolean = !!(this.studentTables && this.studentTables.length);
-      const hasFbSessionData: boolean = !!(this.fbSessionDataTables && this.fbSessionDataTables.length);
-      if (!hasStudents && !hasFbSessionData) {
-        this.statusMessageService.showWarningMessage('No results found.');
-      }
-    }, (resp: ErrorMessageOutput) => {
-      this.statusMessageService.showErrorMessage(resp.error.message);
-    });
+    this.httpRequestService.get('/studentsAndSessionData/search', paramMap)
+        .pipe(finalize(() => this.loadingBarService.hideLoadingBar()))
+        .subscribe((resp: SearchResult) => {
+          this.studentTables = resp.searchStudentsTables;
+          this.fbSessionDataTables = resp.searchFeedbackSessionDataTables;
+          const hasStudents: boolean = !!(this.studentTables && this.studentTables.length);
```

Comment:
Is this like a double-negative, which is the same as `this.studentTables && this.studentTables.length`?